src/macros/macrorecorder.cpp

```diff
@@ -46,23 +51,24 @@ bool MacroRecorder::claimRecording() {
     return m_macroRecordingState.compare_exchange_weak(armed, MacroState::Recording);
 }
 
-void pollRecordingStart(MacroRecorder* pMacroRecorder) {
-    while (pMacroRecorder->getActiveChannel() == nullptr) {
-        QThread::msleep(300);
-        if (pMacroRecorder->getState() == MacroState::Disabled) {
+void MacroRecorder::pollRecordingStart() {
+    qCDebug(macros) << "Polling for recording start";
+    if (getState() != MacroState::Disabled) {
+        if (getActiveChannel() == nullptr) {
             return;
+        } else {
+            ControlProxy(ConfigKey(kMacroRecordingKey, "recording_status")).set(2);
```

Comment:
What is this magic number 2? static_cast the enum class to an int

Reply:
Well, the Macro recording state behaves a bit different from the status now - so while I could reuse it, it wouldn't be good style I think. Hmm...

Reply:
If they don't have the same meaning, then make a separate enum class. But don't scatter magic numbers throughout the code.

Reply:
Added a TODO :)